javalib/src/main/scala/java/util/concurrent/atomic/Striped64.scala

```diff
@@ -0,0 +1,288 @@
+// Ported from JSR 166 revision 1.28
+
+package java.util.concurrent.atomic
+
+import java.lang.Double._
+import java.util.Arrays;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.DoubleBinaryOperator;
+import java.util.function.LongBinaryOperator;
+import scala.scalanative.annotation._
+import scala.scalanative.unsafe._
+import scala.scalanative.libc.atomic.{CAtomicInt, CAtomicLongLong, memory_order}
+import scala.scalanative.runtime.{fromRawPtr, Intrinsics}
+
+/*
+ * Written by Doug Lea with assistance from members of JCP JSR-166
+ * Expert Group and released to the public domain, as explained at
+ * http://creativecommons.org/publicdomain/zero/1.0/
+ */
+
+@SuppressWarnings(Array("serial"))
+object Striped64 {
+
+  final private[atomic] class Cell private[atomic] (var value: Long) {
+    @volatile private[concurrent] var _value = value
```

Comment:
Why do we need `value` and `_value`? If they need to be separate, we should remove `var` from `value`, otherwise we have two fields instead of one field and a ctor argument..

Also, I think the original Java version has some padding here? I'm not sure if that trick works in Scala Native, but if it does, we should port it as well. If it doesn't, we should figure out a way to support that kind of memory layout, because it's important for concurrent code.

For a discussion about padding and how it prevents "false-sharing" on the JVM, see https://github.com/typelevel/cats-effect/pull/2543#issuecomment-968362106.
> That would create false sharing, i.e. two independent fields residing on the same cache line while different threads independently write to each variable. Each write to either field in this case invalidates the cache of the other thread, slowing everyone down.

Reply:
Relevant JSR166 code.
https://github.com/armanbilge/jsr166/blob/abd155f72243cc2066c31ee732d21894f0edf7bd/src/jsr166e/Striped64.java#L88-L92

Reply:
Sorry, just realized I was looking at an old version :)

However, I see the latest code uses `@jdk.internal.vm.annotation.Contended`. This is an important annotation that relates to the false-sharing issue. @WojciechMazur I wonder if we can port its semantics to Scala Native?

> At the VM’s discretion, those fields annotated as contended are given extra padding so they do not share cache lines with other fields that are likely to be independently accessed. This helps to avoid cache contention across multiple threads.

https://docs.azul.com/prime/Contended

Reply:
I opened an issue to track this.
- https://github.com/scala-native/scala-native/issues/3359